extension/inet/inet_escape_functions.cpp

```diff
@@ -0,0 +1,213 @@
+	      html5_names(ReturnHTML5NameCharrefs()), invalid_codepoints(ReturnInvalidCodepoints()),
+	      invalid_charrefs(ReturnInvalidCharrefs()) {
+	}
+};
+
+unique_ptr<FunctionData> UnescapeBind(ClientContext &context, ScalarFunction &bound_function,
+                                      vector<unique_ptr<Expression>> &arguments) {
+	duckdb_re2::RE2::Options options;
+	options.set_log_errors(false); // disable error logging for performance
+	options.set_case_sensitive(false);
+	return make_uniq<RegexpUnescapeBindData>(options);
+}
+
+static void PrepareStrForTypeCasting(string &match) {
+	// in HTML, the sequence &#<number>; represents a character by its Unicode code point or in the form
+	// &#x<hex_number>; which represents its hexadecimal value.
+	// To apply the TryCast::Operation, this string requires some "cleaning"
+	match.erase(match.begin()); // rmv # char
+	if (match.back() == ';') {  // rmv ; char
+		match.pop_back();
+	}
+
+	// TryCast::Operation needs the hexadecimal values to start with 0x
+	if (match[0] == 'x' || match[0] == 'X') {
+		match.insert(0, "0");
+	}
+}
+
+static bool AddBlob(uint32_t code_point, string &result) {
+	int sz = 0;
+	char c[4] = {'\0', '\0', '\0', '\0'};
+	if (!Utf8Proc::CodepointToUtf8(code_point, sz, c)) {
+		return false;
+	}
+	result += c;
+	return true;
+}
+
+static string ReplaceCharref(string_t &str, RegexpUnescapeBindData &info, RE2 &pattern) {
+	string match;
+	string result {""};
+	idx_t start_pos = 0;
+	auto input_data = str.GetData();
+	auto input_size = str.GetSize();
+	auto input = regexp_util::CreateStringPiece(str); // wrap a StringPiece around it
+	while (RE2::FindAndConsume(&input, pattern, &match)) {
+		// include the input value until the current match
+		idx_t match_pos = input_size - (match.length() + 1) - input.length(); // +1 to include also the & ampersand char
+		result += string(input_data + start_pos, match_pos - start_pos);
+
+		// the position after the end of the current match
+		start_pos = (match_pos + 1) + match.length();
+
+		if (match[0] == '#') { // true, if it represents a hexadecimal value or a Unicode point
+			int32_t num;
+			PrepareStrForTypeCasting(match);
+			if (!TryCast::Operation<string_t, int32_t>(string_t(match), num)) {
+				result += "\uFFFD";
+				continue;
+			}
+
+			if (info.invalid_charrefs.find(num) != info.invalid_charrefs.end()) {
+				auto ch = info.invalid_charrefs.at(num);
+				if (Utf8Proc::Analyze(ch.c_str(), ch.length()) == UnicodeType::INVALID) {
+					auto blob = string_t(info.invalid_charrefs.at(num));
+					auto str_blob = Blob::ToString(blob);
+					result += str_blob;
+				} else {
+					result += ch;
+				}
+
+			} else if ((0xD800 <= num && num <= 0xDFFF) || 0x10FFFF < num) {
+				result += "\uFFFD";
+			} else if (info.invalid_codepoints.find(num) != info.invalid_codepoints.end()) {
+				// skip this character
+				continue;
+			} else {
+				result += static_cast<char>(num);
```

Comment:
Can this also use the `AddBlob` (or `AddCodepoint` after the rename)?

Can we also add some tests for these, for high numbers and invalid numbers, e.g.:

```sql
select html_unescape('&#129414');
```